lib/matplotlib/_api.py

```diff
@@ -0,0 +1,34 @@
+
+
+def check_in_list(_values, *, _print_supported_values=True, **kwargs):
+    """
+    For each *key, value* pair in *kwargs*, check that *value* is in *_values*.
+
+    Parameters
+    ----------
+    _values : iterable
+        Sequence of values to check on.
+    _print_supported_values : bool, default: True
+        Whether to print *_values* when raising ValueError.
+    **kwargs : dict
+        *key, value* pairs as keyword arguments to find in *_values*.
+
+    Raises
+    ------
+    ValueError
+        If any *value* in *kwargs* is not found in *_values*.
+
+    Examples
+    --------
+    >>> _api.check_in_list(["foo", "bar"], arg=arg, other_arg=other_arg)
+    """
+    values = _values
+    for key, val in kwargs.items():
+        if val not in values:
+            if _print_supported_values:
+                raise ValueError(
+                    f"{val!r} is not a valid value for {key}; "
+                    f"supported values are {', '.join(map(repr, values))}")
+            else:
+                raise ValueError(
+                    f"{val!r} is not a valid value for {key}")
```

Comment:
Flake8 *really* wants a blank line at the end of the file apparently.

Reply:
It fits in one line too, I think.